app/src/main/java/comcmput301f16t01/github/carrier/Requests/RiderRequestActivity.java

```diff
@@ -195,24 +211,39 @@ public GeoPoint getCenter() {
 
         Location retLoc = new Location("");
 
-        if(startLat > endLat) {
-            retLoc.setLatitude(endLat + ((startLat - endLat)/2));
+        if (startLat > endLat) {
+            retLoc.setLatitude(endLat + ((startLat - endLat) / 2));
         } else {
-            retLoc.setLatitude(startLat + ((endLat - startLat)/2));
+            retLoc.setLatitude(startLat + ((endLat - startLat) / 2));
         }
 
-        if(startLong > endLong) {
-            retLoc.setLongitude(endLong + ((startLong - endLong)/2));
+        if (startLong > endLong) {
+            retLoc.setLongitude(endLong + ((startLong - endLong) / 2));
         } else {
-            retLoc.setLongitude(startLong + ((endLong - startLong)/2));
+            retLoc.setLongitude(startLong + ((endLong - startLong) / 2));
         }
 
         return new GeoPoint(retLoc);
     }
 
-    // TODO fill in
-    public void payForRequest(View view) {
-        Toast.makeText(activity, "PAY FOR REQUEST", Toast.LENGTH_SHORT).show();
+    /**
+     * Will confirm the completion of a request.
+     * @param view The current view.
+     */
+    public void completeRequest(View view) {
+        if (request.getStatus() == Request.CONFIRMED) {
+            Toast.makeText(activity, "PAY FOR REQUEST", Toast.LENGTH_SHORT).show();
+            RequestController rc = new RequestController();
+            rc.completeRequest(request);
+            ImageView statusImageView = (ImageView) findViewById(R.id.imageView_requestStatus);
+            statusImageView.setImageResource(R.drawable.complete);
+
+        }
+        else {
+            AlertDialog.Builder adb = new AlertDialog.Builder(RiderRequestActivity.this);
+            adb.setTitle("Error:");
+            adb.setMessage("Can not complete request");
```

Comment:
Removed this dialog since the button will not be clickable unless you can complete a request.